File: utils/rnapegjunction2tabix.js
if(process.argv.length!=4) {
	console.log('<input RNApeg output file (1. 1-based junction, 2. count 3. type)> <output file /path/to/basename>')
	process.exit()
}


var infile=process.argv[2],
	outfile=process.argv[3]


var fs=require('fs'),
	exec=require('child_process').execSync


var data=[]
var lines=fs.readFileSync(infile,'utf8').trim().split('\n')
var err=0
for(var i=1; i<lines.length; i++) {
	var l=lines[i].split('\t')
	const readcount=l[1]
	const type=l[2]

	var t=l[0].split(/[:,]/)
	const chr=t[0]
	const strand=t[2]
	var start=parseInt(t[1]),
		stop=parseInt(t[4])
	if(isNaN(start) || isNaN(stop)) {
		err++
		continue
	}
	data.push(
		chr+'\t'+(start-1)+'\t'+(stop-1)+'\t'+strand+'\t'+type+'\t'+readcount
	)
}

if(err) {
	console.error('%d invalid junctions in %s',err,infile)
}


fs.writeFileSync(outfile,data.join('\n'),{encoding:'utf8'})

exec('sort -k1,1 -k2,2n '+outfile+' > '+outfile+'.sort')
exec('mv '+outfile+'.sort '+outfile)
exec('bgzip '+outfile)
exec('tabix -p bed '+outfile+'.gz')

File: utils/rnapegjunction2tabix.py
#!/usr/bin/python3

import sys
import re
import os

if len(sys.argv) == 1:
	print('<input RNApeg output file (1. 1-based junction, 2. count 3. type)> <output file /path/to/basename>')
	sys.exit(1)


infile = sys.argv[1]
outfile = sys.argv[2]


fh = open(infile)
out = open(outfile,'w')
for line in fh:
	if line.startswith('junction'):
		continue
	l = line.strip().split('\t')
	readcount = l[1]
	type = l[2]
	t = re.split('[:,]',l[0])
	chrom = t[0]
	strand = t[2]
	try:
		start = int(t[1])
		stop = int(t[4])
	except:
		print('invalid junction: '+line,file=sys.stderr)
		continue
	out.write('\t'.join([chrom,str(start-1),str(stop-1),strand,type,readcount])+'\n')
fh.close()
out.close()

os.system('sort -k1,1 -k2,2n '+outfile+' > '+outfile+'.sort')
os.system('mv '+outfile+'.sort '+outfile)
os.system('bgzip '+outfile)
os.system('tabix -p bed '+outfile+'.gz')
